### ISSUE TITLE
Error loading plugin 'fastlane-plugin-firebase_app_distribution' in 0.4.0

### ISSUE THREAD

**Issue by iRYO400** [fastlane/fastlane-plugin-firebase_app_distribution#293]
### Issue Description
I run fastlane only on Gitlab CI and yesterday it stopped working.
Fastlane is imported using just two words `gem "fastlane"` inside `Gemfile`, same for plugin `gem 'fastlane-plugin-firebase_app_distribution'` inside `Pluginfile`.

But once I set it to fixed version `0.3.8`, it started working

##### Complete verbose output

```
... 
Fetching fastlane-plugin-firebase_app_distribution 0.4.0
... 
Installing fastlane-plugin-firebase_app_distribution 0.4.0
...
[08:27:45]: Error loading plugin 'fastlane-plugin-firebase_app_distribution': cannot load such file -- fastlane/plugin/firebase_app_distribution
[08:27:45]: It seems like you wanted to load some plugins, however they couldn't be loaded
[08:27:45]: Please follow the troubleshooting guide: https://docs.fastlane.tools/plugins/plugins-troubleshooting/
+-------------------------------------------+-----------+------------------+
|                               Used plugins                               |
+-------------------------------------------+-----------+------------------+
| Plugin                                    | Version   | Action           |
+-------------------------------------------+-----------+------------------+
| fastlane-plugin-firebase_app_distribution | undefined | No actions found |
+-------------------------------------------+-----------+------------------+
[!] No actions were found while loading one or more plugins
    Please use `bundle exec fastlane` with plugins
    More info - https://docs.fastlane.tools/plugins/using-plugins/#run-with-plugins
```

| Library     | Version |
| ----------- | ----------- |
| fastlane | [the latest] |
| fastlane-plugin-firebase_app_distribution | [the latest]

**Comment by rebehe** [comment]
Thanks for filing the ticket! I wasn't able to reproduce it on my end. I'm using fastlane version 2.209.1, and I tried updating my Pluginfile to use both no specified version (`gem 'fastlane-plugin-firebase_app_distribution'`) and specifying the latest version (`gem 'fastlane-plugin-firebase_app_distribution', "0.4.0"`) - and both worked for me.

A few troubleshooting ideas:
- Can you try running the command with the `--verbose` flag?
- Can you send your fastlane version info and your Pluginfile setup?

**Comment by iRYO400** [comment]
Yes, for sure.

Here is Gemfile:
```
source "https://rubygems.org"

gem "fastlane"

plugins_path = File.join(File.dirname(__FILE__), 'fastlane', 'Pluginfile')
eval_gemfile(plugins_path) if File.exist?(plugins_path)
```

Pluginfile:
```
# Autogenerated by fastlane
#
# Ensure this file is checked in to source control!

gem 'fastlane-plugin-firebase_app_distribution'
```

And the gitlab job:
```
build_debug_and_deploy:
  stage: build_debug
  image: $REGISTRY_URL/$CI_PROJECT_NAME:v0.0.1
  only:
    - master
    - develop
    - merge_requests
  before_script:
    - bundle install
  script:
    - bundle exec fastlane test --verbose
    - bundle exec fastlane buildDebugAndDeploy
  artifacts:
    paths:
      - app/build/outputs/
    expire_in: 3 days
  allow_failure: false
```


Beginning of the logs with --verbose:
```
DEBUG [2023-01-21 04:18:17.04]: Checking if there are any plugins that should be loaded...
DEBUG [2023-01-21 04:18:17.04]: Loading 'fastlane-plugin-firebase_app_distribution' plugin
INFO [2023-01-21 04:18:17.04]: [32mgem 'fastlane-plugin-firebase_app_distribution' is already installed[0m
ERROR [2023-01-21 04:18:17.04]: [31mError loading plugin 'fastlane-plugin-firebase_app_distribution': cannot load such file -- fastlane/plugin/firebase_app_distribution[0m
DEBUG [2023-01-21 04:18:17.05]: uninitialized constant Fastlane::FirebaseAppDistribution
ERROR [2023-01-21 04:18:17.05]: [31mIt seems like you wanted to load some plugins, however they couldn't be loaded[0m
ERROR [2023-01-21 04:18:17.05]: [31mPlease follow the troubleshooting guide: https://docs.fastlane.tools/plugins/plugins-troubleshooting/[0m
+-------------------------------------------+-----------+------------------+
|                               [32mUsed plugins[0m                               |
+-------------------------------------------+-----------+------------------+
| Plugin                                    | Version   | Action           |
+-------------------------------------------+-----------+------------------+
| [31mfastlane-plugin-firebase_app_distribution[0m | undefined | [31mNo actions found[0m |
+-------------------------------------------+-----------+------------------+

[31m[!] No actions were found while loading one or more plugins[0m
[31m    Please use `bundle exec fastlane` with plugins[0m
[31m    More info - https://docs.fastlane.tools/plugins/using-plugins/#run-with-plugins[0m

[33mSuccessfully loaded Appfile at path '/builds/[Hidden]/fastlane/Appfile'[0m
-------
```

And the ending:
```
+------+------------------+-------------+
|           fastlane summary            |
+------+------------------+-------------+
| Step | Action           | Time (in s) |
+------+------------------+-------------+
| 1    | default_platform | 0           |
| 2    | test             | 626         |
+------+------------------+-------------+
Looking for related GitHub issues on fastlane/fastlane...
Search query: Malformed version number string undefined
URL: https://api.github.com/search/issues?q=Malformed%20version%20number%20string%20undefined+repo:fastlane/fastlane
➡️  java.lang.IllegalArgumentException: Unknown package:
    https://github.com/fastlane/fastlane/issues/18527 [closed] 4 💬
    19 Aug 2021
➡️  snapshot works locally, but not on CircleCI
    https://github.com/fastlane/fastlane/issues/17193 [closed] 10 💬
    25 Dec 2020
➡️  version.rb Malformed version number string undefined (ArgumentError)
    https://github.com/fastlane/fastlane/issues/15715 [closed] 4 💬
    28 Jan 2020
and 3 more at: [https://github.com/fastlane/fastlane/search?q=Malformed%20version%20number%20string%20undefined&type=Issues&utf8=✓](https://github.com/fastlane/fastlane/search?q=Malformed%20version%20number%20string%20undefined&type=Issues&utf8=%E2%9C%93)
/usr/lib/ruby/2.7.0/rubygems/version.rb:215:in `initialize': Malformed version number string undefined (ArgumentError)
	from /usr/lib/ruby/2.7.0/rubygems/version.rb:206:in `new'
	from /usr/lib/ruby/2.7.0/rubygems/version.rb:206:in `new'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:29:in `block in show_update_status'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:26:in `each'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:26:in `show_update_status'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:45:in `ensure in start'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:55:in `start'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
/usr/lib/ruby/2.7.0/rubygems/version.rb:215:in `initialize': \e[31m[!] Malformed version number string undefined\e[0m (ArgumentError)
	from /usr/lib/ruby/2.7.0/rubygems/version.rb:206:in `new'
	from /usr/lib/ruby/2.7.0/rubygems/version.rb:206:in `new'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:29:in `block in show_update_status'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:26:in `each'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:26:in `show_update_status'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/lane_manager_base.rb:20:in `finish_fastlane'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/lane_manager.rb:67:in `cruise_lane'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
/usr/lib/ruby/2.7.0/rubygems/version.rb:215:in `initialize': Malformed version number string undefined (ArgumentError)
	from /usr/lib/ruby/2.7.0/rubygems/version.rb:206:in `new'
	from /usr/lib/ruby/2.7.0/rubygems/version.rb:206:in `new'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:29:in `block in show_update_status'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:26:in `each'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/plugins/plugin_update_manager.rb:26:in `show_update_status'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/lane_manager_base.rb:20:in `finish_fastlane'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/lane_manager.rb:67:in `cruise_lane'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /var/lib/gems/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /var/lib/gems/2.7.0/gems/fastlane-2.211.0/bin/fastlane:23:in `<top (required)>'
	from /usr/local/bin/fastlane:23:in `load'
	from /usr/local/bin/fastlane:23:in `<main>'
Cleaning up project directory and file based variables
```

**Comment by rebehe** [comment]
@iRYO400 hey, it seems like I may need some more info. Not sure what's going on, but maybe we can debug together. Can you join the Firebase community slack channel https://firebase.community/ and DM me?

**Comment by mrtnrst** [comment]
@rebehe also hitting the same issue but with GH actions (opposed to gitlab here). Happy to send over more info if need be.

My gem/plugin file/ runner job are pretty much the same as @iRYO400.

**Comment by anandsathe67** [comment]
Might have something to do with the version of Ruby installed. Try Ruby 3.1.2 @iRYO400 @rebehe

**Comment by iRYO400** [comment]
> @iRYO400 hey, it seems like I may need some more info. Not sure what's going on, but maybe we can debug together. Can you join the Firebase community slack channel https://firebase.community/ and DM me?

It looks like we work in opposite time zones, so I'll try @anandsathe67’s assumption first

**Comment by iRYO400** [comment]
> Might have something to do with the version of Ruby installed. Try Ruby 3.1.2 @iRYO400 @rebehe

I run `ruby -v`  command on CI and got next:
```
ruby 2.7.0p0 (2019-12-25 revision 647ee6f091) [x86_64-linux-gnu]
```

Is it too old? @anandsathe67

**Comment by anandsathe67** [comment]
I had a bunch of trouble with 2.7 @iRYO400 - things have worked smoothly since I went to 3.1 (currently on 3.1.2)
GIve it a shot if it isnt too much trouble

**Comment by anandsathe67** [comment]
ruby -v
`ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin20]`

**Comment by iRYO400** [comment]
@anandsathe67 thanks, I'll try it. But before I have to tell to my devops

**Comment by mrtnrst** [comment]
@anandsathe67 / @iRYO400 I have been running this on ruby 3.0.2p107, noted by [GH Actions](https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md). Looking at the blame for the file, it looks like it hasn't changed in 8 months so unsure if ruby would be the root cause here.

@rebehe there are a couple of old issues that allude to the same issue output but I don't understand enough the Fastlane plugin system to know if they would be related. e.g. #43 or #15

**Comment by rebehe** [comment]
Thanks @mrtnrst, [#15](http://go/gh/fastlane/fastlane-plugin-firebase_app_distribution/issues/15) looks particularly related, especially given the error message you're seeing is the one that was added because of that thread ([PR](http://go/gh/fastlane/fastlane/pull/15532)).

Can't vouch for any personally, but just collecting a few of the recommendations in that thread here:
- Upgrade your ruby version
- Make sure you're using `bundle exec fastlane` instead of just `fastlane` for your command
- Run `bundle install --path vendor/bundle` and then use `bundle exec fastlane`
- Run `sudo fastlane install_plugins`
- If on Mac, it could be a file permission issue. Edit the permission for the gem `sudo chmod -R a+r /Library/Ruby/Gems/X.X.X./gems/fastlane-plugin-firebase_app_distribution-0.4.0`. Alternatively, you could run `sudo gem uninstall fastlane-plugin-firebase_app_distribution` and then `gem install fastlane-plugin-firebase_app_distribution --user-install` to reinstall

**Comment by rebehe** [comment]
We think we found the issue, and shipped a new version (`0.4.2`) to fix this. Can you all try again and let us know if it's fixed? @iRYO400 @mrtnrst

**Comment by iRYO400** [comment]
> We think we found the issue, and shipped a new version (`0.4.2`) to fix this. Can you all try again and let us know if it's fixed? @iRYO400 @mrtnrst

🥳 seems like it fixes the issue without specifying the concrete version
